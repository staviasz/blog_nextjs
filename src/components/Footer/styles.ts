import styled, { css } from 'styled-components';
import { Text } from '../Text/styles';

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    ${Text} {
      font-size: ${theme.font.sizes.small};
      text-align: center;
      margin: 0;
    }
    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
