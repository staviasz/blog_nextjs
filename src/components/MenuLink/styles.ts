import styled, { css } from 'styled-components';

export const MenuLink = styled.a`
  ${({ theme }) => css`
    display: block;
    color: ${theme.colors.white};
    text-decoration: none;
    margin-bottom: ${theme.spacings.small};
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 150ms ease-in-out;

    &:hover {
      border-right-color: ${theme.colors.secondary};
      color: ${theme.colors.secondary};
    }
  `}
`;

export const MenuDiv = styled.div`
  ${({ theme }) => css`
    display: block;
    color: ${theme.colors.white};
    text-decoration: none;
    margin-bottom: ${theme.spacings.small};
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 150ms ease-in-out;

    &:hover {
      border-right-color: ${theme.colors.secondary};
      color: ${theme.colors.secondary};
    }
  `}
`;
