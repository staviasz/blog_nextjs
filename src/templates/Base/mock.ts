import { BaseTemplateProps } from '.';
import json from '../../api/data.json';
import { mapSettings } from '../../utils/mapData/mapSettings';

const { data } = json;
export const mock = {
  settings: mapSettings(data.setting),
  children: 'Ol√° mundo',
} as BaseTemplateProps;
