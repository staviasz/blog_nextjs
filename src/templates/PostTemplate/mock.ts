import { PostTemplateProps } from '.';
import json from '../../api/data.json';
import mapPosts from '../../utils/mapData/mapPost';
import { mapSettings } from '../../utils/mapData/mapSettings';

const { data } = json;
export default {
  settings: mapSettings(data.setting),
  post: mapPosts(data.posts)[0],
} as PostTemplateProps;
