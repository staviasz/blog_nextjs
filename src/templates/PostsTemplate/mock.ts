import { PostsTemplateProps } from '.';
import json from '../../api/data.json';
import mapPosts from '../../utils/mapData/mapPost';
import { mapSettings } from '../../utils/mapData/mapSettings';

const { data } = json;
export default {
  settings: mapSettings(data.setting),
  posts: mapPosts(data.posts),
} as PostsTemplateProps;
