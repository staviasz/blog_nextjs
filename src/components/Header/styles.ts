import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';
import { LinkContent } from '../LogoLink/styles';

export const Container = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding-bottom: ${theme.spacings.xhuge};
    width: 100%;
    max-width: ${theme.sizes.max};
    color: ${theme.colors.darkText};
    font-size: ${theme.font.sizes.small};
    margin: 0 auto;
    ${HeadingStyles} {
      margin: 0 0 calc(${theme.spacings.small} - 1rem);
    }
    ${LinkContent} {
      > img {
        max-height: 100%;
        width: 12rem;
        height: 12rem;
        border-radius: 50%;
      }
    }
  `}
`;

export const Content = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column wrap;
    margin-left: ${theme.spacings.large};
    max-width: 48rem;
  `}
`;
