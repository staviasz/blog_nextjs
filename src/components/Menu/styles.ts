import styled, { css, DefaultTheme } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  $menuvisible: string;
};

const ContainerChanger = (
  menuvisible: MenuBehaviorProps['$menuvisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuvisible === 'open' ? '0' : '-30rem'};
  overflow-y: ${menuvisible === 'open' ? 'auto' : 'hidden'};

  @media ${theme.media.lteMedium} {
    left: ${menuvisible === 'open' ? '0' : '-32rem'};
  }
`;

export const Container = styled.div<MenuBehaviorProps>`
  ${({ theme, $menuvisible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;
    ${ContainerChanger($menuvisible, theme)}

    img {
      width: calc(${theme.spacings.xhuge} * 3);
      height: calc(${theme.spacings.xhuge} * 3);
      border-radius: 50%;
    }
  `}
`;

export const Nav = styled.nav`
  margin: auto;
  width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xxlarge};

      img {
        border: 0.5rem solid ${theme.colors.secondary};
      }
    }
  `}
`;

const buttonChanger = (
  menuvisible: MenuBehaviorProps['$menuvisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuvisible ? '26rem' : '1rem'};
  color: ${menuvisible ? theme.colors.secondary : theme.colors.white};

  @media ${theme.media.lteMedium} {
    left: ${menuvisible ? '26rem' : '-0.5rem'};
  }
`;

export const OpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, $menuvisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    transition: all 300ms ease-in-out;
    ${buttonChanger($menuvisible, theme)}
  `}
`;
