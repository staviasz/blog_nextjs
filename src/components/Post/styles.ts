import styled, { css } from 'styled-components';
import { Theme } from '../../styles/theme';

interface PostContainerProps {
  size: 'max' | 'content';
}

const postContainerStyles = {
  max: (theme: Theme) => css`
    max-width: ${theme.sizes.max};
  `,
  content: (theme: Theme) => css`
    max-width: ${theme.sizes.content};
  `,
};

export const PostContainer = styled.article<PostContainerProps>`
  ${({ theme, size }) => css`
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    ${postContainerStyles[size](theme)}

    @media ${theme.media.lteMedium} {
      padding: 0 calc(${theme.spacings.large} / 2);
    }
  `}
`;

export const Content = styled.div``;
