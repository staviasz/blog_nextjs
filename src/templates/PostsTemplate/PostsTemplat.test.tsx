import { PostsTemplate, PostsTemplateProps } from '.';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

const props: PostsTemplateProps = mock;

jest.mock('next/router', () => ({
  useRouter: () => ({
    push: jest.fn(),
    replace: jest.fn(),
    prefetch: jest.fn(),
    query: { q: '' },
  }),
}));

describe('<PostsTemplate />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<PostsTemplate {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('should match snapshot without posts', () => {
    const { container } = renderTheme(
      <PostsTemplate {...props} posts={undefined} />,
    );

    expect(container).toMatchSnapshot();
  });
});
