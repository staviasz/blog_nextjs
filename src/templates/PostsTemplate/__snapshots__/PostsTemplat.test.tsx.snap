// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostsTemplate /> should match snapshot 1`] = `
.c11 {
  font-size: 2.4rem;
  text-transform: uppercase;
  font-style: italic;
}

.c16 {
  font-size: 2.4rem;
  text-transform: lowercase;
  font-style: italic;
}

.c22 .c10 {
  margin: 0;
  margin-top: calc(1.4rem - 0.6rem);
  margin-bottom: 1.4rem;
}

.c22 a {
  text-decoration: none;
  color: inherit;
  transition: all 300ms ease-in-out;
}

.c22:hover a {
  color: #dc143c;
}

.c22:hover img {
  opacity: 0.8;
}

.c23 {
  max-width: 100%;
  transition: opacity 300ms ease-in-out;
}

.c20 {
  width: 100%;
}

.c21 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

.c29 {
  font-size: 1.8rem;
  text-align: left;
}

.c29 p {
  margin: 4.0rem 0;
}

.c29 a {
  color: #dc143c;
}

.c29 a:hover {
  filter: brightness(50%);
}

.c29 pre {
  background-color: #000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFF;
}

.c29 img {
  max-width: 100%;
  padding: 4.8rem 0;
}

.c27 {
  border-top: 0.1rem solid #DDD;
}

.c27 .c28 {
  font-size: 1.4rem;
  text-align: center;
  margin: 0;
}

.c27 a {
  color: inherit;
  text-decoration: none;
  font-size: 1.4rem;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c30 {
  position: fixed;
  background: #000;
  color: #FFF;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
}

.c14 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333;
  font-size: 1.4rem;
  margin: 0 auto;
}

.c14 .c10 {
  margin: 0 0 calc(1.4rem - 1rem);
}

.c14 .c10 img {
  max-height: 100%;
  width: 12rem;
  height: 12rem;
  border-radius: 50%;
}

.c15 {
  display: flex;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c12 {
  display: block;
  color: #FFF;
  text-decoration: none;
  margin-bottom: 1.4rem;
  border-right: 0.5rem solid #000;
  transition: all 150ms ease-in-out;
}

.c12:hover {
  border-right-color: #dc143c;
  color: #dc143c;
}

.c7 {
  background: #000;
  padding: 3.2rem;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -30rem;
  overflow-y: hidden;
}

.c7 img {
  width: calc(6.4rem * 3);
  height: calc(6.4rem * 3);
  border-radius: 50%;
}

.c8 {
  margin: auto;
  width: 100%;
}

.c9 .c10 {
  display: flex;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c9 .c10 img {
  border: 0.5rem solid #dc143c;
}

.c5 {
  position: fixed;
  top: 2.4rem;
  color: #FFF;
  background: #000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  transition: all 300ms ease-in-out;
  left: 1rem;
  color: #FFF;
}

.c0 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0,0,0,0);
}

.c2 {
  appearance: none;
  opacity: 0;
  width: 0;
  height: 0;
}

.c2:checked+.c3 {
  background: #dc143c;
}

.c2:focus+.c3 {
  box-shadow: 0 0 1px #dc143c;
}

.c2:checked+.c3:before {
  transform: translateX(1.9rem);
  background: #000;
}

.c4 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #DDD;
  transition: all 300ms ease-in-out;
  border-radius: 2rem;
  box-shadow: 0 0 1px #000;
}

.c4:before {
  content: '';
  position: absolute;
  height: 2.1rem;
  width: 2.1rem;
  left: 0.4rem;
  bottom: 0.2rem;
  background: #333;
  border-radius: 50%;
  transition: all 300ms ease-in-out;
}

.c13 {
  margin-top: 6.4rem;
}

.c19 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c26 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

.c17 {
  margin: 0 auto;
  margin-bottom: 4.0rem;
  padding: 0 3.2rem;
  max-width: 120rem;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.c18 {
  padding: 0.5rem 1.4rem;
}

.c24 {
  padding: 0 3.2rem;
  margin: 3.2rem 0;
  display: flex;
  justify-content: center;
}

.c25 {
  background: #000;
  color: #FFF;
  border: none;
  padding: 1.4rem 3.2rem;
  cursor: pointer;
}

.c25:disabled {
  background: #AAAAAA;
}

@media (max-width: 500px) {
  .c21 {
    grid-template-columns: 1fr;
  }
}

@media (max-width: 500px) {
  .c14 .c10 {
    display: flex;
    flex-flow: column wrap;
  }

  .c14 .c10>.c10 {
    margin: 0 0 2.4rem 0;
  }
}

@media (max-width: 500px) {
  .c15 {
    margin-left: 0;
  }
}

@media (max-width: 768px) {
  .c7 {
    left: -32rem;
  }
}

@media (max-width: 768px) {
  .c5 {
    left: -0.5rem;
  }
}

<div>
   
  <div
    class=""
  >
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Toggle light and dark modes
        <input
          class="c2"
          type="checkbox"
          value="Dark mode active"
        />
        <span
          class="c3 c4"
        />
      </label>
    </div>
    <a
      aria-label="Open or close menu"
      class="c5"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c6"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c7"
    >
      <nav
        class="c8"
      >
        <div
          class="c9"
        >
          <h1
            class="c10 c11"
          >
            <a
              href="/"
              target="_self"
            >
              <img
                alt="Tecnologia"
                src="https://res.cloudinary.com/drko8mcj9/image/upload/v1691590342/desenvolvimento_web_73903_164_e1eaca8f8a.webp"
              />
            </a>
          </h1>
        </div>
        <a
          class="c12"
          href="#"
          target="_self"
        >
          desenvolvimento web
        </a>
      </nav>
    </div>
    <div
      class="c13"
    >
      <header
        class="c14"
      >
        <h1
          class="c10 c11"
        >
          <a
            href="/"
            target="_self"
          >
            <img
              alt="Tecnologia - tecnologia em geral"
              src="https://res.cloudinary.com/drko8mcj9/image/upload/v1691590342/desenvolvimento_web_73903_164_e1eaca8f8a.webp"
            />
          </a>
        </h1>
        <div
          class="c15"
        >
          <h2
            class="c10 c16"
          >
            Tecnologia
          </h2>
          <p>
            tecnologia em geral
          </p>
        </div>
      </header>
    </div>
    <div
      class="c17"
    >
      <form
        action="/search/"
        method="GET"
      >
        <input
          class="c18"
          name="q"
          placeholder="Encontre posts"
          type="search"
          value=""
        />
      </form>
    </div>
    <div
      class="c19"
    >
      <div
        class="c20"
      >
        <div
          class="c21"
        >
          <div
            class="c22"
          >
            <h2
              class="c10 c16"
            >
              <a
                href="/senioridade"
              >
                Junior, pleno e senior
              </a>
            </h2>
            <p
              class=""
            >
              Senioridade em desenvolvimento de sofware junior, pleno e senior
            </p>
          </div>
          <div
            class="c22"
          >
            <a
              href="/desenvolvimento-web"
            >
              <img
                alt="desenvolvimento web"
                class="c23"
                src="https://res.cloudinary.com/drko8mcj9/image/upload/v1691449461/photo_1498050108023_c5249f4df085_df84325049.webp"
              />
            </a>
            <h2
              class="c10 c16"
            >
              <a
                href="/desenvolvimento-web"
              >
                desenvolvimento web
              </a>
            </h2>
            <p
              class=""
            >
              As varias areas de desenvolvimento web
            </p>
          </div>
        </div>
      </div>
      <div
        class="c24"
      >
        <button
          class="c25"
        >
          Carregar mais
        </button>
      </div>
    </div>
    <div
      class="c26"
    >
      <footer
        class="c27"
      >
        <p
          class="c28 c29"
        >
          Projeto criado por Erick Staviasz
        </p>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c30"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c6"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
   
</div>
`;

exports[`<PostsTemplate /> should match snapshot without posts 1`] = `
.c11 {
  font-size: 2.4rem;
  text-transform: uppercase;
  font-style: italic;
}

.c16 {
  font-size: 2.4rem;
  text-transform: lowercase;
  font-style: italic;
}

.c20 {
  width: 100%;
}

.c21 {
  padding: 3.2rem;
  text-align: center;
  font-size: 1.4rem;
}

.c22 {
  max-width: 120rem;
  margin: 0 auto;
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(28rem,1fr));
  gap: 3.2rem;
  padding: 3.2rem;
}

.c26 {
  font-size: 1.8rem;
  text-align: left;
}

.c26 p {
  margin: 4.0rem 0;
}

.c26 a {
  color: #dc143c;
}

.c26 a:hover {
  filter: brightness(50%);
}

.c26 pre {
  background-color: #000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFF;
}

.c26 img {
  max-width: 100%;
  padding: 4.8rem 0;
}

.c24 {
  border-top: 0.1rem solid #DDD;
}

.c24 .c25 {
  font-size: 1.4rem;
  text-align: center;
  margin: 0;
}

.c24 a {
  color: inherit;
  text-decoration: none;
  font-size: 1.4rem;
}

.c6 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c27 {
  position: fixed;
  background: #000;
  color: #FFF;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
}

.c14 {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-bottom: 6.4rem;
  width: 100%;
  max-width: 96rem;
  color: #333;
  font-size: 1.4rem;
  margin: 0 auto;
}

.c14 .c10 {
  margin: 0 0 calc(1.4rem - 1rem);
}

.c14 .c10 img {
  max-height: 100%;
  width: 12rem;
  height: 12rem;
  border-radius: 50%;
}

.c15 {
  display: flex;
  flex-flow: column wrap;
  margin-left: 3.2rem;
  max-width: 48rem;
}

.c12 {
  display: block;
  color: #FFF;
  text-decoration: none;
  margin-bottom: 1.4rem;
  border-right: 0.5rem solid #000;
  transition: all 150ms ease-in-out;
}

.c12:hover {
  border-right-color: #dc143c;
  color: #dc143c;
}

.c7 {
  background: #000;
  padding: 3.2rem;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -30rem;
  overflow-y: hidden;
}

.c7 img {
  width: calc(6.4rem * 3);
  height: calc(6.4rem * 3);
  border-radius: 50%;
}

.c8 {
  margin: auto;
  width: 100%;
}

.c9 .c10 {
  display: flex;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c9 .c10 img {
  border: 0.5rem solid #dc143c;
}

.c5 {
  position: fixed;
  top: 2.4rem;
  color: #FFF;
  background: #000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  transition: all 300ms ease-in-out;
  left: 1rem;
  color: #FFF;
}

.c0 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0,0,0,0);
}

.c2 {
  appearance: none;
  opacity: 0;
  width: 0;
  height: 0;
}

.c2:checked+.c3 {
  background: #dc143c;
}

.c2:focus+.c3 {
  box-shadow: 0 0 1px #dc143c;
}

.c2:checked+.c3:before {
  transform: translateX(1.9rem);
  background: #000;
}

.c4 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #DDD;
  transition: all 300ms ease-in-out;
  border-radius: 2rem;
  box-shadow: 0 0 1px #000;
}

.c4:before {
  content: '';
  position: absolute;
  height: 2.1rem;
  width: 2.1rem;
  left: 0.4rem;
  bottom: 0.2rem;
  background: #333;
  border-radius: 50%;
  transition: all 300ms ease-in-out;
}

.c13 {
  margin-top: 6.4rem;
}

.c19 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c23 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

.c17 {
  margin: 0 auto;
  margin-bottom: 4.0rem;
  padding: 0 3.2rem;
  max-width: 120rem;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.c18 {
  padding: 0.5rem 1.4rem;
}

@media (max-width: 500px) {
  .c22 {
    grid-template-columns: 1fr;
  }
}

@media (max-width: 500px) {
  .c14 .c10 {
    display: flex;
    flex-flow: column wrap;
  }

  .c14 .c10>.c10 {
    margin: 0 0 2.4rem 0;
  }
}

@media (max-width: 500px) {
  .c15 {
    margin-left: 0;
  }
}

@media (max-width: 768px) {
  .c7 {
    left: -32rem;
  }
}

@media (max-width: 768px) {
  .c5 {
    left: -0.5rem;
  }
}

<div>
   
  <div
    class=""
  >
    <div
      class="c0"
    >
      <label
        class="c1"
      >
        Toggle light and dark modes
        <input
          class="c2"
          type="checkbox"
          value="Dark mode active"
        />
        <span
          class="c3 c4"
        />
      </label>
    </div>
    <a
      aria-label="Open or close menu"
      class="c5"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c6"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c7"
    >
      <nav
        class="c8"
      >
        <div
          class="c9"
        >
          <h1
            class="c10 c11"
          >
            <a
              href="/"
              target="_self"
            >
              <img
                alt="Tecnologia"
                src="https://res.cloudinary.com/drko8mcj9/image/upload/v1691590342/desenvolvimento_web_73903_164_e1eaca8f8a.webp"
              />
            </a>
          </h1>
        </div>
        <a
          class="c12"
          href="#"
          target="_self"
        >
          desenvolvimento web
        </a>
      </nav>
    </div>
    <div
      class="c13"
    >
      <header
        class="c14"
      >
        <h1
          class="c10 c11"
        >
          <a
            href="/"
            target="_self"
          >
            <img
              alt="Tecnologia - tecnologia em geral"
              src="https://res.cloudinary.com/drko8mcj9/image/upload/v1691590342/desenvolvimento_web_73903_164_e1eaca8f8a.webp"
            />
          </a>
        </h1>
        <div
          class="c15"
        >
          <h2
            class="c10 c16"
          >
            Tecnologia
          </h2>
          <p>
            tecnologia em geral
          </p>
        </div>
      </header>
    </div>
    <div
      class="c17"
    >
      <form
        action="/search/"
        method="GET"
      >
        <input
          class="c18"
          name="q"
          placeholder="Encontre posts"
          type="search"
          value=""
        />
      </form>
    </div>
    <div
      class="c19"
    >
      <div
        class="c20"
      >
        <p
          class="c21"
        >
          Nenhum post encontrado aqui =( 
        </p>
        <div
          class="c22"
        />
      </div>
    </div>
    <div
      class="c23"
    >
      <footer
        class="c24"
      >
        <p
          class="c25 c26"
        >
          Projeto criado por Erick Staviasz
        </p>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c27"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c6"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
   
</div>
`;
